sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -297,6 +297,7 @@ case class PreprocessTableCreation(catalog: SessionCatalog) extends Rule[Logical
 
     val normalizedPartCols = normalizePartitionColumns(schema, table)
     val normalizedBucketSpec = normalizeBucketSpec(schema, table)
+    val normalizedClusterBySpec = normalizeClusterBySpec(schema, table)
```

Comment:
instead of normalizing it here, shall we normalize it before we convert cluster spec to table properties?

Reply:
done